src/graph-helpers/deserialize-object.js

```diff
@@ -0,0 +1,59 @@
+import descriptorForField from './descriptor-for-field';
+import ClassRegistry from './class-registry';
+
+function extractDescriptors(objectGraph, typeName) {
+  return Object.keys(objectGraph).map(fieldName => {
+    return descriptorForField(fieldName, typeName);
+  });
+}
+
+function isSingularRelationship(descriptor) {
+  return !(descriptor.typeName === 'Scalar' || descriptor.isList);
```

Comment:
Oh! Hah! It's like a made a recyclable predicate but didn't actually do anything with it!

Will fix 😄